cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -657,4 +661,51 @@ private void writeHistory(Id.NamespacedId targetId) {
     byte[] row = MdsHistoryKey.getMdsKey(targetId, System.currentTimeMillis()).getKey();
     indexedTable.put(row, Bytes.toBytes(HISTORY_COLUMN), Bytes.toBytes(GSON.toJson(metadata)));
   }
+
+  /**
+   * Rebuilds all the indexes in the {@link MetadataDataset} in batches.
+   *
+   * @param startRow the row to start the scan for the current batch
+   * @param limit the batch size
+   * @return the row key of the last row scanned in the current batch, {@code null} if there are no more rows to scan.
+   */
+  @Nullable
+  public Row rebuildIndexes(@Nullable Row startRow, int limit) {
```

Comment:
We could just return the rowkey as byte[] instead of the complete row, and same for the parameter startRow.